appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockTranslator.java

```diff
@@ -0,0 +1,123 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2022 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.client.editor.simple.components;
+
+import com.google.appinventor.client.DesignToolbar;
+import com.google.appinventor.client.Ode;
+import com.google.appinventor.client.OdeAsyncCallback;
+import com.google.appinventor.client.editor.simple.SimpleEditor;
+import com.google.appinventor.client.widgets.properties.EditableProperty;
+
+import com.google.gwt.user.client.ui.Image;
+import com.google.gwt.user.client.ui.Widget;
+
+/**
+ * Mock for the non-visible Translator component. This needs a separate mock
+ * from other non-visible components so that we can fetch the token property
+ * from the server
+ *
+ * based on {@link MockCloudDB}.
```

Comment:
This seems like a weird linebreak.

Reply:
Yep, but intentional. I guess I should change "based" to "Based" so it is clear